src/main/java/bsh/BSHArrayDimensions.java

```diff
@@ -117,7 +117,15 @@ public Object eval( CallStack callstack, Interpreter interpreter )
             {
                 try {
                     Object length = jjtGetChild(i).eval(callstack, interpreter);
-                    definedDimensions[i] = ((Primitive)length).intValue();
+                    // Expect this to be a primitive, but also unbox numbers
+                    int dim;
+                    if (length instanceof Integer ||
+                        length instanceof Short ||
+                        length instanceof Byte)
```

Comment:
Test for `length instanceof Number` to also include the other number types Float, Double, etc.